src/v/storage/disk_log_impl.cc

```diff
@@ -777,9 +777,6 @@ ss::future<> disk_log_impl::gc(compaction_config cfg) {
       "[{}] applying 'deletion' log cleanup policy with config: {}",
       config().ntp(),
       cfg);
-    if (unlikely(cfg.asrc->abort_requested())) {
-        return ss::make_ready_future<>();
-    }
```

Comment:
Might be more future proof to just leave this in. At least, it isn't obvious without some digging that this gets inlined, and it seems like that'd be an easy fact to change.

Reply:
not inlined like c++ inline, i mean it is called with no discernible scheduling points between it and the log manager driving it removal.

Reply:
right. i think the issue is that it doesn't seem like a property to be broken. it's more like an optimization that wasn't effective given the context.